### PR TITLE
Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)

### PR DESCRIPTION
**Pull request by orosz-usgs**
New mppublications/sipp endpoint in MpPublicationMvcService
  New pojo for json posted: SippMpPubRequest
  Refactored method ISippProcess.processInformationProduct to return MpPublication
  Refactored DisseminationListService.processDisseminationList to have th try/catch inside the for loop.
  Redid tests.
PUBSTWO-1661:
 Fixed typo in ORCID_PREFIX

### CODE REVIEW

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java [hunk 1]
```diff
@@ -49,24 +48,30 @@ public SippProcess(
...
-		ProcessSummary rtn = new ProcessSummary();
-		StringBuilder processingDetails = new StringBuilder(ipNumber + ":");
+	public MpPublication processInformationProduct(ProcessType processType, String ipNumber) {
+		MpPublication rtn = null;
 
+		MpPublication mpPublication = null;
 		InformationProduct informationProduct = getInformationProduct(ipNumber);
-		MpPublication mpPublication = getMpPublication(informationProduct);
+
+		if(informationProduct != null) {
```
Comment (dsteinich) [inline src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java:57]: While unexpected, this check is ok.

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java [hunk 2]
```diff
@@ -49,24 +48,30 @@ public SippProcess(
...
+			mpPublication = getMpPublication(informationProduct);
+		}
+
+		if(mpPublication == null) {
+			mpPublication = new MpPublication();
```
Comment (dsteinich) [inline src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java]: However, this instantiation of an MpPublication is not - code in okToProcess (line 173) expects a null mpPublication if none is found on line 58. Either move this if into the else block below (line 68-73), or remove it and instantiate the MpPublication in line 71 (rtn = ...)

File: src/test/java/gov/usgs/cida/pubs/webservice/mp/MpPublicationMvcServiceTest.java [hunk 3]
```diff
@@ -97,7 +100,7 @@
 
 	@Before
 	public void setup() {
-		mvcService = new MpPublicationMvcService(pubBusService, busService);
+		mvcService = new MpPublicationMvcService(pubBusService, busService, sippService);
```
Comment (dsteinich) [inline src/test/java/gov/usgs/cida/pubs/webservice/mp/MpPublicationMvcServiceTest.java:110]: I'd like to see some testing of the new end point in here. At the least, to show that the mock service gets called and to see that the posted parameters are validated.

File: src/main/java/gov/usgs/cida/pubs/springinit/SecurityConfig.java [hunk 4]
```diff
@@ -47,23 +47,23 @@ public void configure(ResourceServerSecurityConfigurer resources) {
...
-				.authorizeRequests()
-					//anonymous (public)
-					.antMatchers(HttpMethod.OPTIONS, "/**").permitAll()
-					.antMatchers("/lookup/**", "/publication/**", "/version", "/about/**").permitAll()
-					.antMatchers("/swagger-ui.html", "/webjars/**", "/swagger-resources/**", "/v2/api-docs", "/v3/api-docs").permitAll()
-					//authenticated
-					.antMatchers("/mppublications/*/preview*", "/auth/logout").fullyAuthenticated()
-					//authorized
-					.antMatchers("/**").hasAnyAuthority(configurationService.getAuthorizedAuthorities())
+			.and().authorizeRequests().antMatchers("/**").permitAll();
```
Comment (dsteinich) [inline src/main/java/gov/usgs/cida/pubs/springinit/SecurityConfig.java]: We should probably create two different security configs and activate them via spring profiles so we don't accidentally turn of security in production.

**Review by dsteinich — CHANGES_REQUESTED** [review]

File: src/main/java/gov/usgs/cida/pubs/springinit/SecurityConfig.java (hunk 4 above)
Comment (orosz-usgs) [reply src/main/java/gov/usgs/cida/pubs/springinit/SecurityConfig.java]: agreed, the other option would be to figure out Authorization in a dev environment.

File: src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java [hunk 5]
```diff
@@ -217,35 +223,30 @@ protected ProcessSummary processPublication(ProcessType processType, Information
...
-			notOkDetails.append(" PublicationSubtype: ").append(informationProduct.getPublicationSubtype().getText());
-		}
-		if (null != informationProduct.getUsgsSeriesTitle()) {
-			notOkDetails.append(" Series: ").append(informationProduct.getUsgsSeriesTitle().getText());
+		String notProcessedMess = String.format("IPDS record for IPNumber '%s' not processed", ipNumber);
+
+		StringBuilder notOkDetails = new StringBuilder("\n\t").append(notProcessedMess).append(" (\"").append(processType).append("\") -")
+				.append(" ProductType: ").append(productType);
+
+		if(informationProduct != null) {
```
Comment (mbucknell) [inline src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java]: I'm surprised Codacy didn't flag this. Should there be a space between **if** and **(**?

**Review by mbucknell — APPROVED** [review]

**Review by dsteinich — APPROVED** [review]

**Pull request merged**